### PR TITLE
Checklist 1st view

### PR DESCRIPTION
**Pull request by vietanh253**

### CODE REVIEW

File: DaiQiOS/DaiQiOS/Todaychecklist.swift [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import SwiftUI
+
+struct ChecklistItem {
+    var title: String
+    var description: String
+    var isChecked: Bool = false
+}
```
Comment (knn90) [inline DaiQiOS/DaiQiOS/Todaychecklist.swift:7]: Please move ChecklistItem to separate file

File: DaiQiOS/DaiQiOS/Todaychecklist.swift [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    var body: some View{
+        ZStack {
+                   Color.white.edgesIgnoringSafeArea(.all)
+                   
+                   Todaychecklist()
+                       .background(Color.white)
+                       .padding()
+                     }
+    }
+}
```
Comment (knn90) [inline DaiQiOS/DaiQiOS/Todaychecklist.swift]: ContainerView should be a subview of Todaychecklist. 
Like:
`TodayCheckList {
    Container{
         ZStack
    }
}`

File: DaiQiOS/DaiQiOS/Todaychecklist.swift [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                    .listStyle(PlainListStyle())
+                
+            
+    }
+    
+    struct Todaychecklist_Previews: PreviewProvider {
+        static var previews: some View {
+            Container()
+        }
+    }
```
Comment (knn90) [inline DaiQiOS/DaiQiOS/Todaychecklist.swift:75]: The Preview should be a separate struct

File: DaiQiOS/DaiQiOS/Todaychecklist.swift [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+                   Color.white.edgesIgnoringSafeArea(.all)
+                   
+                   Todaychecklist()
+                       .background(Color.white)
+                       .padding()
+                     }
+    }
+}
+
+struct Todaychecklist: View {
```
Comment (knn90) [inline DaiQiOS/DaiQiOS/Todaychecklist.swift:21]: Naming convention: TodayChecklist

**Review by knn90 — CHANGES_REQUESTED** [review]

File: DaiQiOS/DaiQiOS/Todaychecklist.swift [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+struct TodayChecklist: View {
+    @State private var checklistItems: [ChecklistItem] = [
+        ChecklistItem(title: "Task 1", description: "Description 1"),
+        ChecklistItem(title: "Task 2", description: "Description 2"),
+        ChecklistItem(title: "Task 3", description: "Description 3")
+    ]
+    
+    var body: some View {
+        ZStack {
+            //            Color.white.edgesIgnoringSafeArea(.all)
```
Comment (knn90) [inline DaiQiOS/DaiQiOS/Todaychecklist.swift:12]: Please remove commented code

File: DaiQiOS/DaiQiOS/Todaychecklist.swift [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+                    }
+                }
+                Button(action: {}) {
+                    Image(systemName: "plus")
+                        .font(.system(size: 24))
+                        .frame(width: 64, height: 64)
+                        .background(Color.purple.opacity(0.3))
+                        .foregroundColor(.white)
+                        .clipShape(Circle())
+                }
```
Comment (knn90) [inline DaiQiOS/DaiQiOS/Todaychecklist.swift:47]: Please use @ViewBuilder to extract this button with name `addTaskButton`

File: DaiQiOS/DaiQiOS/Todaychecklist.swift [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+                            
+                            VStack(alignment: .leading, spacing: 4) {
+                                Text(checklistItems[index].title)
+                                    .font(.headline)
+                                
+                                Text(checklistItems[index].description)
+                                    .font(.subheadline)
+                                    .foregroundColor(.secondary)
+                            }
+                        }
```
Comment (knn90) [inline DaiQiOS/DaiQiOS/Todaychecklist.swift:37]: Create a new view for example `TaskCell`  to make the `TodayChecklist` shorter

File: DaiQiOS/DaiQiOS/Todaychecklist.swift [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+            //            Color.white.edgesIgnoringSafeArea(.all)
+            VStack{
+                Text("Quest")
+                    .padding(20)
+                    .frame(alignment: .center)
+                
+                List {
+                    Text("Today Checklist")
+                        .font(.title)
+                        .bold()
```
Comment (knn90) [inline DaiQiOS/DaiQiOS/Todaychecklist.swift:21]: Make this a navigationBarTitle instead of a Text

File: DaiQiOS/DaiQiOS/Todaychecklist.swift [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+        ChecklistItem(title: "Task 3", description: "Description 3")
+    ]
+    
+    var body: some View {
+        ZStack {
+            //            Color.white.edgesIgnoringSafeArea(.all)
+            VStack{
+                Text("Quest")
+                    .padding(20)
+                    .frame(alignment: .center)
```
Comment (knn90) [inline DaiQiOS/DaiQiOS/Todaychecklist.swift:16]: We can remove this

**Review by knn90 — CHANGES_REQUESTED** [review]

File: DaiQiOS/DaiQiOS/Todaychecklist.swift (hunk 8 above)
Comment (vietanh253) [reply DaiQiOS/DaiQiOS/Todaychecklist.swift:23]: navigationBarTitle is deprecated

File: DaiQiOS/DaiQiOS/TaskCell.swift [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+var checklistItems: [ChecklistItem] = [
+        ChecklistItem(title: "Task 1", description: "Description 1"),
+        ChecklistItem(title: "Task 2", description: "Description 2"),
+        ChecklistItem(title: "Task 3", description: "Description 3")
+    ]
+    
+struct TaskCell: View {
+    var body: some View {
+        
+        ForEach(checklistItems.indices, id: \.self) { index in
```
Comment (knn90) [inline DaiQiOS/DaiQiOS/TaskCell.swift:13]: Move ForEach to the List

File: DaiQiOS/DaiQiOS/Todaychecklist.swift [hunk 11]
```diff
@@ -0,0 +1,41 @@
+import SwiftUI
+struct TodayChecklist: View {
+var body: some View {
+        ZStack {
+            VStack{
+                NavigationView {
+                   Text("Today Checklist")
+    }
```
Comment (knn90) [inline DaiQiOS/DaiQiOS/Todaychecklist.swift:8]: Fix indentation

File: DaiQiOS/DaiQiOS/Todaychecklist.swift [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+                .navigationTitle("Today Checklist")  .frame(height: 50)
+                .font(.title)
+                .bold()
+                            List {
```
Comment (knn90) [inline DaiQiOS/DaiQiOS/Todaychecklist.swift:12]: Fix indentation

**Review by knn90 — CHANGES_REQUESTED** [review]

**Review by knn90 — APPROVED** [review]

**Pull request merged**